.github/workflows/eden-trusted.yml

```diff
@@ -175,18 +176,73 @@ jobs:
             }
 
 
+  select_eden_version:
+    name: Select Eden Version
+    runs-on: ubuntu-latest
+    needs: context
+    if: needs.context.outputs.skip_run == 'false'
+    outputs:
+      workflow_ref: ${{ steps.select.outputs.workflow_ref }}
+      eden_version: ${{ steps.select.outputs.eden_version }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
+
+      - name: Select Eden version based on target branch
+        id: select
+        env:
+          BASE_REF: ${{ needs.context.outputs.pr_base_ref }}
+        run: |
+          MAPPING_FILE=".github/eden-version-map.yml"
+
+          if [[ ! -f "$MAPPING_FILE" ]]; then
+            echo "::error::Eden version mapping file not found: $MAPPING_FILE"
+            exit 1
+          fi
+
+          echo "PR target branch: $BASE_REF"
+
+          # Try to get the mapping for the specific branch
+          workflow_ref=$(yq eval ".\"$BASE_REF\".workflow_ref" "$MAPPING_FILE")
+          eden_version=$(yq eval ".\"$BASE_REF\".eden_version" "$MAPPING_FILE")
+
+          # If no specific mapping exists, use the default
+          if [[ "$workflow_ref" == "null" || -z "$workflow_ref" ]]; then
+            echo "No specific mapping found for branch '$BASE_REF', using default"
+            workflow_ref=$(yq eval '.default.workflow_ref' "$MAPPING_FILE")
+            eden_version=$(yq eval '.default.eden_version' "$MAPPING_FILE")
+          fi
+
+          # Verify that we have valid values
+          if [[ "$workflow_ref" == "null" || -z "$workflow_ref" ]]; then
+            echo "::error::Failed to determine workflow_ref for branch '$BASE_REF'"
+            exit 1
+          fi
+
+          if [[ "$eden_version" == "null" || -z "$eden_version" ]]; then
+            echo "::error::Failed to determine eden_version for branch '$BASE_REF'"
+            exit 1
+          fi
+
+          echo "Selected Eden workflow: $workflow_ref"
+          echo "Selected Eden version: $eden_version"
+
+          echo "workflow_ref=$workflow_ref" >> "$GITHUB_OUTPUT"
+          echo "eden_version=$eden_version" >> "$GITHUB_OUTPUT"
+
+
   tests:
     name: ${{ needs.context.outputs.eden_parent_job_title }}
-    needs: context
+    needs: [context, select_eden_version]
     if: needs.context.outputs.skip_run == 'false'
-    uses: lf-edge/eden/.github/workflows/test.yml@1.0.13
+    uses: ${{ needs.select_eden_version.outputs.workflow_ref }}
```

Comment:
the linter tells me it's not gonna work: GH action needs a static link to the workflow.